app/upload/views.py

```diff
@@ -72,3 +82,22 @@ def upload_document(service_id):
             'mimetype': mimetype,
         }
     ), 201
+
+
+def _upload_document_with_hashed_recipient_email(verification_email, service_id, file_data, mimetype):
+    hasher = PasswordHasher(
+        memory_cost=15360,
+        time_cost=2,
+        parallelism=1,
+        hash_len=16,
+        type=Type.ID
+    )
+
+    hashed_recipient_email = hasher.hash(verification_email)
+    document = document_store.put(
+        service_id,
+        file_data,
+        mimetype=mimetype,
+        metadata={"Hashed-Recipient-Email": hashed_recipient_email}
+    )
+    return document
```

Comment:
I think it might make more sense to have this as part of the DocumentStore so that stuff outside of the store doesn't have to care about hashing (and how to hash) the email address.

It might be cleaner if we just pass the email address to the `document_store.put` call, and then that can manage hashing the email (if provided) and storing it on the document. What do you think?

Reply:
sure thing, will move it there ☺️

Reply:
hmmm should we have a call about this? I am not sure I understand.